lib/mastodon/emoji_cli.rb

```diff
@@ -72,13 +73,68 @@ def import(path)
       say("Imported #{imported}, skipped #{skipped}, failed to import #{failed}", color(imported, skipped, failed))
     end
 
+    option :category
+    option :shortcode, type: :boolean
+    option :overwrite, type: :boolean
+    desc 'export PATH', 'Export emoji to a TAR GZIP archive at PATH'
+    long_desc <<-LONG_DESC
+      Exports custom emoji to 'export.tar.gz' at PATH.
+
+      The --category option dumps only the specified category. If the given category doesn't exist, all emoji will be exported.
+
+      The --overwrite option will overwrite an existing archive.
+    LONG_DESC
+
+    def export(path)
+      exported = 0
+      skipped = 0
+      failed = 0
+      category = options[:category] ? CustomEmojiCategory.find_by(name: options[:category]) : nil
+
+      export_file_name = File.join(path, 'export.tar.gz')
+      if options[:overwrite] || !File.file?(export_file_name)
+        File.open(export_file_name, 'wb') do |file|
+          Zlib::GzipWriter.wrap(file) do |gzip|
+            Gem::Package::TarWriter.new(gzip) do |tar|
+              if category.nil?
+                say('Dumping all emoji...')
+                CustomEmoji.local.all? do |emoji|
+                  emoji_file_name = export_file_name(emoji)
+                  say("Adding '#{emoji.shortcode}' as '#{emoji_file_name}'...")
+                  tar.add_file_simple(emoji_file_name, 0o644, emoji.image_file_size) do |io|
+                    io.write Paperclip.io_adapters.for(emoji.image).read
+                    exported += 1
```

Comment:
This is duplicated code and could probably be moved in a function.

Reply:
👌